File: ProfitPilot/Data/Models/Trade.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProfitPilot.Data.Models
{
    public class Trade
    {
        [Key]
        public int TradeId { get; set; }

        [Required]
        public string UserId { get; set; } // Foreign key to ApplicationUser

        [ForeignKey("UserId")]
        public ApplicationUser User { get; set; }

        [Required]
        public string Asset { get; set; } // e.g., BTCUSDT, ISCTR

        [Required]
        public DateTime EntryTime { get; set; }

        [Required]
        public decimal EntryPrice { get; set; }

        public DateTime? ExitTime { get; set; }

        public decimal? ExitPrice { get; set; }

        public decimal Quantity { get; set; }

        public string TradingStyle { get; set; } // e.g., Scalping, Swing

        public decimal? ProfitLoss { get; set; }

        public string Status { get; set; } // Open or Closed

        // Additional properties can be added as needed
    }
}

File: ProfitPilotServer/src/main/java/SWE599/ProfitPilot/trade/Trade.java
package SWE599.ProfitPilot.trade;

import SWE599.ProfitPilot.user.User;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "trades")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Trade {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String asset; // e.g., BTCUSDT, ISCTR

    private Double entryPrice;

    private Double exitPrice;

    private Double quantity;

    private Double profitLoss; // Calculated as (exitPrice - entryPrice) * quantity

    private LocalDateTime entryTime;

    private LocalDateTime exitTime;

    private String tradeStyle; // e.g., scalp, swing

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;
}